Makefile

```diff
@@ -37,60 +39,107 @@ CRD_OPTIONS ?= "crd:crdVersions=v1"
 
 CONTROLLER_GEN=go run sigs.k8s.io/controller-tools/cmd/controller-gen
 
+##@ General
+
+# The help target prints out all targets with their descriptions organized
+# beneath their categories. The categories are represented by '##@' and the
+# target descriptions by '##'. The awk commands is responsible for reading the
+# entire set of makefiles included in this invocation, looking for lines of the
+# file as xyz: ## something, and then pretty-format the target and help. Then,
+# if there's a line with ##@ something, that gets pretty-printed as a category.
+# More info on the usage of ANSI control characters for terminal formatting:
+# https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_parameters
+# More info on the awk command:
+# http://linuxcommand.org/lc3_adv_awk.php
+
+.PHONY: help
+help: ## Display this help.
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_0-9-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
 .PHONY: all
-all: generate manifests controller verify
+all: generate manifests controller verify ##
+
+##@ Development
 
-# Build controller binary
 .PHONY: controller
-controller: generate fmt vet
+controller: generate fmt vet ## Build controller binary
 	go build -o bin/manager cmd/workcontroller/workcontroller.go
 
-# Run go fmt against code
 .PHONY: fmt
-fmt:
+fmt: ## Run go fmt against code
 	go fmt ./...
 
-# Run go vet against code
 .PHONY: vet
-vet:
+vet: ## Run go vet against code
 	go vet ./...
 
-# Run generators for Deepcopy funcs and CRDs
+##@ Development
+
 .PHONY: generate
-generate:
+generate: ## Run generators for Deepcopy funcs and CRDs
```

Comment:
I wonder are those comments mostly for your personal understanding or something else? I normally don't see comments put here. Maybe above.